rpc/grpc/init.go

```diff
@@ -28,18 +30,39 @@ type MessageReadWriter interface {
 
 type streamWrapper struct {
 	grpc.ClientStream
+	timeout time.Duration
+	cancel  context.CancelFunc
 }
 
 func (w streamWrapper) ReadMessage(m Message) error {
-	return w.ClientStream.RecvMsg(m)
+	return w.withTimeout(func() error {
+		return w.ClientStream.RecvMsg(m)
+	})
 }
 
 func (w streamWrapper) WriteMessage(m Message) error {
-	return w.ClientStream.SendMsg(m)
+	return w.withTimeout(func() error {
+		return w.ClientStream.SendMsg(m)
+	})
 }
 
 func (w *streamWrapper) Close() error {
-	return w.ClientStream.CloseSend()
+	return w.withTimeout(w.ClientStream.CloseSend)
+}
+
+func (w *streamWrapper) withTimeout(closure func() error) error {
```

Comment:
Have you checked the performance impact of this for small messages? E.g. `object.Put` uses stream.

Reply:
Tried it on unary `object.Head`. Without business logic (no signature checks and metabase access, pure gRPC communication over localhost), there is a 13-15% overhead. It is worst case scenario, so I expect it will be much less noticeable in deploy environment. 

```
goos: linux
goarch: amd64
pkg: github.com/nspcc-dev/neofs-node/cmd/example/client
cpu: Intel(R) Core(TM) i7-8550U CPU @ 1.80GHz
BenchmarkHeadRequests
BenchmarkHeadRequests-8   	            9391	    112575 ns/op
BenchmarkHeadRequestsWithDeadline
BenchmarkHeadRequestsWithDeadline-8   	    8246	    127810 ns/op
```